lib/msal-core/samples/react-sample-app/src/AuthProvider.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from "react";
+import {
+    msalApp,
+    requiresInteraction,
+    fetchMsGraph,
+    isIE,
+    GRAPH_ENDPOINTS,
+    GRAPH_SCOPES,
+    GRAPH_REQUESTS
+} from "./auth-utils";
+
+// If you support IE, our recommendation is that you sign-in using Redirect APIs
+const useRedirectFlow = isIE();
+
+export default C =>
+    class AuthProvider extends Component {
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                account: null,
+                error: null,
+                emailMessages: null,
+                graphProfile: null
+            };
+        }
+
+        async acquireToken(request, redirect) {
```

Comment:
nitpick, feel free to ignore, but maybe instead of `AuthProvider.acquireToken` you expose `AuthProvider.acquireMailToken` and `AuthProvider.acquireMeToken`  That way the caller doesn't have to construct a request

Reply:
never mind I see that this is onlu called internally